plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.model;
+
+import static org.opensearch.ml.settings.MLCommonsSettings.ML_COMMONS_MONITORING_REQUEST_COUNT;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.model.MLModelState;
+import org.opensearch.ml.engine.Predictable;
+import org.opensearch.ml.profile.MLModelProfile;
+import org.opensearch.ml.profile.MLPredictRequestStats;
+
+@Log4j2
+public class MLModelCacheHelper {
+    private final Map<String, MLModelCache> modelCaches;
+    private volatile Long maxRequestCount;
```

Comment:
This is for debugging purpose in case we need to heavy load test, long could be more flexible. It can be int, but I don't see any big overhead by using long here.